kernel/src/process_load.rs

```diff
@@ -0,0 +1,791 @@
+        match load_result {
+            Ok((new_flash, new_mem, proc)) => {
+                remaining_flash = new_flash;
+                remaining_memory = new_mem;
+                if proc.is_some() {
+                    if config::CONFIG.debug_load_processes {
+                        proc.map(|p| debug!("Loaded process {}", p.get_process_name()));
+                    }
+                    procs[index] = proc;
+                    index = index + 1;
+                }
+            }
+            Err((_new_flash, _new_mem, err)) => {
+                if config::CONFIG.debug_load_processes {
+                    debug!("No more processes to load: {:?}.", err);
+                }
+                // No more processes to load.
+                break;
+            }
+        }
+    }
+    Ok(())
+}
+
+/// Use `checker` to transition `procs` from the `Unverified` into the
+/// `Unstarted` state (if they pass the checker policy) or
+/// `CredentialsFailed` state (if they do not pass the checker
+/// policy). If `verifier` is `None` then all processes are
+/// automatically verified. Processes that transition into the
+/// `Unstarted` state are started by enqueueing a stack frame to run
+/// the initialization function as indicated in the TBF header.
+#[inline(always)]
+fn check_processes(
+    procs: &'static [Option<&'static dyn Process>],
+    kernel: &'static Kernel,
+) -> Result<(), ProcessLoadError> {
+    let capability = create_capability!(ProcessApprovalCapability);
+
+    if kernel.verifier().is_none() {
+        if config::CONFIG.debug_process_credentials {
+            debug!("Checking: no checker provided, load and run all processes");
+        }
+        for proc in procs.iter() {
+            let res = proc.map(|p| {
+                p.mark_credentials_pass(None, None, &capability)
+                    .or(Err(ProcessLoadError::InternalError))?;
+                if config::CONFIG.debug_process_credentials {
+                    debug!("Running {}", p.get_process_name());
+                }
+                kernel
+                    .submit_process(p)
+                    .or(Err(ProcessLoadError::InternalError))?;
+                Ok(())
+            });
+            if let Some(Err(e)) = res {
+                return Err(e);
+            }
+        }
+        Ok(())
+    } else {
+        kernel
+            .verifier()
+            .map_or(Err(ProcessLoadError::InternalError), |v| {
+                #[allow(unused_mut)] // machine doesn't need mut
+                let machine = unsafe {
+                    static_init!(
+                        ProcessCheckerMachine,
+                        ProcessCheckerMachine {
+                            process: Cell::new(0),
+                            footer: Cell::new(0),
+                            verifier: OptionalCell::empty(),
+                            processes: procs,
+                            kernel: kernel
+                        }
+                    )
```

Comment:
If this is ok, it at least needs a comment explaining the exception to our general "no static init outside of board crates".